demisto_sdk/commands/pre_commit/pre_commit_command.py

```diff
@@ -0,0 +1,288 @@
+            )
+            if not find_path_index:
+                raise Exception(f"Could not find integration/script path for {file}")
+            integration_script_path = Path(*file.parts[: next(find_path_index) + 1])
+            integrations_scripts_mapping[integration_script_path].add(file)
+        else:
+            infra_files.append(file)
+
+    python_versions_to_files = defaultdict(set)
+    with multiprocessing.Pool() as pool:
+        integrations_scripts = pool.map(
+            BaseContent.from_path, integrations_scripts_mapping.keys()
+        )
+
+    for integration_script in integrations_scripts:
+        if not integration_script or not isinstance(
+            integration_script, IntegrationScript
+        ):
+            continue
+        integration_script_path = integration_script.path.parent
+        if python_version_string := integration_script.python_version:
+            version = Version(python_version_string)
+            python_version_string = f"{version.major}.{version.minor}"
+        python_versions_to_files[
+            python_version_string or DEFAULT_PYTHON2_VERSION
+        ].update(
+            integrations_scripts_mapping[integration_script_path]
+            | {integration_script.path}
+        )
+
+    python_versions_to_files[DEFAULT_PYTHON_VERSION].update(infra_files)
+    return python_versions_to_files
+    return PreCommitRunner(python_versions_to_files)
+
+
+def pre_commit_manager(
+    input_files: Optional[Iterable[Path]] = None,
+    use_git: bool = False,
+    all_files: bool = False,
+    test: bool = False,
+    skip_hooks: Optional[List[str]] = None,
+    validate: bool = False,
+    format: bool = False,
+    native_images: bool = False,
+    verbose: bool = False,
+    show_diff_on_failure: bool = False,
+) -> int:
+    """Run pre-commit hooks .
+
+    Args:
+        input_files (Iterable[Path], optional): Input files to run pre-commit on. Defaults to None.
+        use_git (bool, optional): Whether use git to determine precommit files. Defaults to False.
+        all_files (bool, optional): Whether to run on all_files. Defaults to False.
+        test (bool, optional): Whether to run unit-tests. Defaults to False.
+        skip_hooks (Optional[List[str]], optional): List of hooks to skip. Defaults to None.
+        force_run_hooks (Optional[List[str]], optional): List for hooks to force run. Defaults to None.
+        verbose (bool, optional): Whether run pre-commit in verbose mode. Defaults to False.
+        show_diff_on_failure (bool, optional): Whether show git diff after pre-commit failure. Defaults to False.
+
+    Returns:
+        int: Return code of pre-commit.
+    """
+    # We have imports to this module, however it does not exists in the repo.
+    (CONTENT_PATH / "CommonServerUserPython.py").touch()
+
+    if not any((input_files, use_git, all_files)):
+        logger.debug("No input_files, use_git, all_files: running on staged files only")
```

Comment:
By default, this will not show when running in CLI. It will only be written to the log file.